processing/src/main/java/io/druid/query/MetricsEmittingExecutorService.java

```diff
@@ -55,21 +50,41 @@ protected ListeningExecutorService delegate()
   @Override
   public <T> ListenableFuture<T> submit(Callable<T> tCallable)
   {
-    emitMetrics();
     return delegate.submit(tCallable);
   }
 
   @Override
   public void execute(Runnable runnable)
   {
-    emitMetrics();
     delegate.execute(runnable);
   }
 
-  private void emitMetrics()
+  private void emitMetrics(ServiceEmitter emitter, ServiceMetricEvent.Builder metricBuilder)
   {
     if (delegate instanceof PrioritizedExecutorService) {
       emitter.emit(metricBuilder.build("segment/scan/pending", ((PrioritizedExecutorService) delegate).getQueueSize()));
     }
   }
+
+  public static class ExecutorServiceMonitor extends AbstractMonitor
+  {
+    private final ExecutorService exec;
+    private final ServiceMetricEvent.Builder metricBuilder;
+
+    @Inject
+    public ExecutorServiceMonitor(@Processing ExecutorService exec)
+    {
+      this.exec = exec;
+      this.metricBuilder = new ServiceMetricEvent.Builder();
+    }
+
+    @Override
+    public boolean doMonitor(ServiceEmitter emitter)
+    {
+      if (exec instanceof MetricsEmittingExecutorService) {
```

Comment:
`MetricsEmittingExecutorService` could accept an `ExecutorServiceMonitor` in its constructor and call `monitor.add(this)`, rather than the other way around.